src/backend/aspen/api/views/auth.py

```diff
@@ -26,23 +27,121 @@
 router = APIRouter()
 
 
+def get_invitation_ticket(
+    a0: Auth0Client, organization: str, invitation: str
+) -> Optional[Auth0OrgInvitation]:
+    org: Auth0Org = {
+        "id": organization,
+        "name": organization,
+        "display_name": organization,
+    }
+    # We can't query the auth0 org invitations api directly if we only have a ticket id,
+    # so this is the next best thing.
+    tickets = a0.get_org_invitations(org)
+    for ticket in tickets:
+        expires = ticket["expires_at"]
+        # Don't process expired invitations
+        if datetime.fromisoformat(expires.rstrip("Z")) < datetime.now():
+            continue
+        if ticket.get("ticket_id") == invitation:
+            return ticket
+    return None
+
+
+async def get_invitation_redirect(
+    oauth: StarletteOAuth2App,
+    settings: Settings,
+    a0: Auth0Client,
+    db: AsyncSession,
+    request: Request,
+    invitation: str,
+    organization: str,
+    organization_name: Optional[str] = None,
+    cookie_userid: Optional[str] = None,
+) -> Optional[Response]:
+    # If this invitation is for an email address that already exists in our db,
+    # we'll use our custom invitation acceptance flow to associate their existing
+    # account with the invitation. If we haven't seen that email address before,
+    # we'll send them to the standard auth0 "create a new account" flow.
+
+    # Load more information about the invitation from auth0
+    ticket_info = get_invitation_ticket(a0, organization, invitation)
+    if not ticket_info:
+        return None
```

Comment:
when would there be no ticket_info?

Reply:
If the invitation they tried to use doesn't exist or is expired.